packages/scene-composer/src/utils/dataBindingUtils.ts

```diff
@@ -160,25 +160,22 @@ export const ruleEvaluator = (
   for (let index = 0; index < rule.statements.length; index++) {
     try {
       const statement = rule.statements[index];
-
       let escapedExpression = statement.expression;
       Object.keys(escapedKeyMap).forEach((key) => {
         escapedExpression = escapedExpression.replace(key, escapedKeyMap[key]);
       });
       const evalResult = jexl.evalSync(escapedExpression, escapedValues);
-
       if (statement.target === IotTwinMakerNumberNamespace) {
         if (isFinite(Number(evalResult))) {
           return Number(evalResult);
         }
       } else if (evalResult) {
-        return statement.target;
+        return { target: statement.target, targetMetadata: statement.targetMetadata };
       }
     } catch (error) {
       // It might be a user mistake for one rule, but the next rule may work, we should continue
       LOG.error('Failed to evaluate rule', error as Error);
     }
   }
-
-  return defaultState;
+  return { target: defaultState, targetMetadata: undefined };
```

Comment:
can you add a return type to this function so the consumers know clearly the type of the value returned? right now I can see 2 returns `{ target: xxx, targetMetadata: yyy}` but the rest return different types. it will be hard to handle the results in this way.
also instead of `{ target: xxx, targetMetadata: yyy}`, I suggest `{ result: xxx, targetMetadata: yyy}` so it makes more sense for cases returning defaultState or number.

Reply:
There are total 5 return results here. We can return targetMetadata if the type of target is icon. But outermost return will return the default target state if nothing is matched but I just added undefined because we need this to evaluate visual state.

Reply:
Returning a RuleReturnResult type.